lib/apiservers/engine/backends/container.go

```diff
@@ -1607,6 +1601,60 @@ func (c *Container) containerAttach(name string, ca *backend.ContainerAttachConf
 	return nil
 }
 
+// ContainerRename changes the name of a container, using the oldName
+// to find the container. An error is returned if newName is already
+// reserved.
+func (c *Container) ContainerRename(oldName, newName string) error {
+	defer trace.End(trace.Begin(newName))
+
+	if oldName == "" || newName == "" {
+		return fmt.Errorf("neither old nor new names may be empty")
+	}
+
+	if !utils.RestrictedNamePattern.MatchString(newName) {
+		return fmt.Errorf("Invalid container name (%s), only %s are allowed", newName, utils.RestrictedNameChars)
+	}
+
+	var err error
+
+	// Look up the container name in the metadata cache to get long ID
+	vc := cache.ContainerCache().GetContainer(oldName)
+	if vc == nil {
+		log.Errorf("Container %s not found", oldName)
```

Comment:
Since you return error all in almost all cases, sometimes you log it, sometimes don't. Can't caller take care about logging the error?